### ISSUE TITLE
Parameter Issue

### ISSUE THREAD

**Issue by srinivas-anant** [Esri/arcgis-cookbook#125]
I am trying to pass only the cookbook and the recipe in the run-list & nothing else. It throws the following error:

`
You must supply a name when declaring a file resource
`

For the recipe portal. I have created a wrapper cookbook which has the following recipes:
- include_recipe 'arcgis-enterprise::system'
- include_recipe 'esri-iis'
- include_recipe 'arcgis-enterprise::portal'
- include_recipe 'arcgis-enterprise::portal_wa'
- include_recipe 'arcgis-enterprise::post_install'

The wrapper cookbook has a attribute file which has all the necessary parameters. But for the node['arcgis']['server']['authorization_file'] it throws an error when I don't mention it in the run-list but when I mention only the above parameter in the run-list the cookbook converges. 

Here is the run-list with the parameter:
```
{ 
 "arcgis": {
    "portal": {
      "authorization_file":"D:\\ArcGIS_License\\Portal.prvc"
    }
},
"run_list": [
 "recipe[arcgis-portal-windows::default]" 
  ] 
} 
```

Here is the run-list without the parameter:
```
{ 
"run_list": [    
 "recipe[arcgis-portal-windows::default]" 
  ] 
} 
```

When I use chef-client to converge the cookbook the run-list with the parameter converges but the run-list without the parameter fails to converge. Why is this so?

**Comment by pbobov** [comment]
Try to set  default['arcgis']['portal']['cached_authorization_file'] and default['arcgis']['cache_authorization_files'] = false  in the attributes file of the wrapper cookbook.

**Comment by pbobov** [comment]
And make sure that default['arcgis']['portal']['authorization_file'] is set in the attributes file of the wrapper cookbook.

**Comment by srinivas-anant** [comment]
Thank you!

**Comment by srinivas-anant** [comment]
The above method you suggested doesn't seem to be working, it is not authorizing the portal servers at all and in my wrapper cookbook I am downloading the setup files and the license files and I am storing in a particular folder all those seem to be omitted and the convergence starts directly from the the include_recipe part. Why is this happening?

**Comment by srinivas-anant** [comment]
This is the error I am getting:

`

    TypeError
    ---------
    no implicit conversion of false into String
`

**Comment by mschoelen** [comment]
What version of Chef-client are you using?

**Comment by srinivas-anant** [comment]
Chef Client -12.19.36

**Comment by srinivas-anant** [comment]
Anybody know how to solve this error?

**Comment by mschoelen** [comment]
I was getting a similar error, and it was resolved by bootstrap installing Chef-client 12.21.3.
